doc/changelog.rst

```diff
@@ -1,12 +1,20 @@
 Changelog
 =========
 
+Version 2.4.0 (2022-07-xx)
+--------------------------
```

Comment:
@pablogsal @vstinner 

Would you like to take look at the changelog entries for proper explanations?